src/app/[locale]/page.tsx

```diff
@@ -24,110 +26,124 @@ const Profile = () => {
     WebProfileApi.getProfile()
       .then((res) => {
         setProfileData(res);
+        if (res === 404) {
+          setProfileIsAvailable(false);
+        } else {
+          setProfileIsAvailable(true);
+        }
       })
       .catch((err) => {
-        console.log(err);
-      });
-
-    WebProfileApi.getUserSessionState()
-      .then((res) => {
-        setSessionData(res);
-      })
-      .catch((err) => {
-        console.log(err);
+        console.log('status', err);
       });
   }, []);
 
-  return (
-    <Grid sx={commonBackground}>
-      <Introduction
-        title={t('common.hello', { nome: userFromStorage?.name })}
-        summary={t('profile.anagraphicinfo')}
-        summaryColumns={{
-          xs: 12,
-          md: 8,
-        }}
-      />
-
-      <Typography variant="h5" my={2}>
-        {t('common.yourprofile')}
-      </Typography>
-      <Grid container gap={2} mb={2} flexWrap={{ xs: 'wrap', sm: 'nowrap', md: 'nowrap' }}>
-        <Grid item xs={12} sm={6} md={6} bgcolor={bgColor}>
-          <Grid padding={3}>
-            <Typography variant="body2">{t('common.namesurname')}</Typography>
-            <Typography variant="sidenav">{`${userFromStorage?.name} ${userFromStorage?.surname}`}</Typography>
-          </Grid>
-          <Divider />
-          <Grid padding={3}>
-            <Typography variant="body2">{t('common.emailaddress')}</Typography>
-            <Typography variant="sidenav">{profileData?.email}</Typography>
-          </Grid>
-        </Grid>
+  useEffect(() => {
+    if (profileIsAvailable) {
+      WebProfileApi.getUserSessionState()
+        .then((res) => {
+          setSessionData(res);
+        })
+        .catch((err) => {
+          console.log(err);
+        });
+    }
+  }, [profileIsAvailable]);
 
-        <Grid item xs={12} sm={6} md={6} bgcolor={bgColor}>
-          <Grid container>
-            <Grid item xs={10} padding={3}>
-              <Typography variant="body2">{t('common.appaccess')}</Typography>
-              <Typography variant="sidenav">
-                {sessionData?.access_enabled ? t('common.abled') : t('common.locked')}
-              </Typography>
-            </Grid>
-            <Grid item xs={2} textAlign={'center'} alignSelf={'center'}>
-              <Tooltip
-                title={
-                  sessionData?.access_enabled
-                    ? t('common.tooltipaccessabled')
-                    : 'L’accesso a IO è bloccato con livello di sicurezza 2'
-                }
-                placement="top"
-                arrow
-              >
-                <HelpOutlineIcon color="primary" />
-              </Tooltip>
+  return (
+    <>
+      {profileIsAvailable === false ? (
```

Comment:
Do we usually do the explicit check for boolean? I mean, why not use `!profileIsAvailable`

Reply:
Thought it was easier to read. Changed the code !isProfileAvailable.